crates/cargo-bsp/src/cargo_communication/request_handle.rs

```diff
@@ -43,8 +78,8 @@ impl RequestHandle {
             cargo_handle,
             cancel_receiver,
         );
-        let thread = jod_thread::Builder::new().spawn(move || actor.run())?;
-
+        let thread =
+            jod_thread::Builder::new().spawn(move || run_commands(actor, requested_cmd))?;
```

Comment:
workflow tego jest dla mnie trochę niejasny...
Może skoro cargo handle i tak jest opcjonalne to można by stworzyć `RequestActora `bez żadnego `CargoHandle`, następnie wywołać na nim funkcje która zrobi to spawnowanie procesu dla  `unit_graph_command` i ogarnie cały ten proces po czym zwróci `UnitGraphStatusCode`. Potem w zależności od tego kodu możemy zachować się tak jak w funkcji `run_commands` (czyli zespawnujemy odpowiednią komendę jeżeli wszystko będzie ok, zakończymy wpp)